### PR TITLE
add annotations of Particle3D

### PR DESCRIPTION
**Pull request by songchengjiang**

### CODE REVIEW

**Comment by songchengjiang** [comment]
@super626

File: extensions/Particle3D/CCParticle3DAffector.h [hunk 1]
```diff
@@ -33,17 +33,47 @@ NS_CC_BEGIN
...
 
+/**
+* @addtogroup _3d
+* @{
+*/
+
+/** @class Particle3DAffector
+*@brief @~english Particle3DAffector is a base class, define interfaces.
+ Specific implementation decisions by different affectors.
+ Affector is mainly bound particles behavior after launch, such as acceleration, rotation, etc.
```
Comment (super626) [inline extensions/Particle3D/CCParticle3DAffector.h]: after being launch

File: extensions/Particle3D/CCParticle3DEmitter.h [hunk 2]
```diff
@@ -32,22 +32,55 @@ NS_CC_BEGIN
...
 /**
- * 3d particle emitter
- */
+* @addtogroup _3d
+* @{
+*/
+
+/** @class Particle3DEmitter
+*@brief @~english Particle3DEmitter is a base class, define interfaces.
+ Specific implementation decisions by different emitters.
```
Comment (super626) [inline extensions/Particle3D/CCParticle3DEmitter.h]: Different emitters can have their own implementation

File: extensions/Particle3D/CCParticle3DEmitter.h [hunk 3]
```diff
@@ -32,22 +32,55 @@ NS_CC_BEGIN
...
+ Emitter is mainly determined how particles emitted, such as point emission, spherical emission, etc.
```
Comment (super626) [inline extensions/Particle3D/CCParticle3DEmitter.h]: Emitter mainly decides how particles are emitted, such as ...

File: extensions/Particle3D/CCParticle3DRender.h [hunk 4]
```diff
@@ -44,33 +44,96 @@ class VertexBuffer;
...
 /**
- * 3d particle render
- */
+* @addtogroup _3d
+* @{
+*/
+
+/** @class Particle3DRender
+*@brief @~english Particle3DRender is a base class, define interfaces.
+ Specific implementation decisions by different renderers.
```
Comment (super626) [inline extensions/Particle3D/CCParticle3DRender.h]: Similar to above file. There is no verbs in the sentence

File: extensions/Particle3D/CCParticle3DRender.h [hunk 5]
```diff
@@ -44,33 +44,96 @@ class VertexBuffer;
...
+ Renderer is mainly determined how particles rendered, such as billboard rendering, model rendering, etc.
```
Comment (super626) [inline extensions/Particle3D/CCParticle3DRender.h]: Renderer mainly determines how particles are rendered,

File: extensions/Particle3D/CCParticleSystem3D.h [hunk 6]
```diff
@@ -138,13 +198,28 @@ class CC_DLL DataPool
...
+    PoolIterator _releasedIter;///< the iterater of allocated pool
+    PoolList _released;///< allocated pool
+    PoolList _locked;///< unallocated pool
 };
 
 typedef DataPool<Particle3D> ParticlePool;
 
+/** @class ParticleSystem3D
+*@brief @~english ParticleSystem3D is a base class, define interfaces.
+ specific implementation decisions by different particle systems.
```
Comment (super626) [inline extensions/Particle3D/CCParticleSystem3D.h]: Similar problem

File: extensions/Particle3D/PU/CCPUEmitter.h [hunk 7]
```diff
@@ -37,147 +37,577 @@
...
 /**
- * 3d particle emitter
- */
+* @addtogroup _3d
+* @{
+*/
+
+/** @class PUEmitter
+*@brief @~english PUEmitter，the base class of pu emitter，inherit from Particle3DEmitter.
+ Specific implementation decisions by different emitters.
```
Comment (super626) [inline extensions/Particle3D/PU/CCPUEmitter.h]: Similar problem

File: extensions/Particle3D/PU/CCPUEmitter.h [hunk 8]
```diff
@@ -365,206 +919,107 @@ class CC_DLL PUEmitter : public Particle3DEmitter
...
-    Vec3 _upVector;
+    PUDynamicAttribute* _dynParticleDepth;///<Dynamic attribute used to define the (own) depth of a particle.
+    bool _dynParticleDepthSet;///<Whether use dynamic attribute to define the depth of a particle.
 
-    /** Particles that are left to be emitted from the previous time.
-    */
-    float _remainder;
+    Vec3 _upVector;///<Notional up vector, just used to speed up generation of variant directions.
+
+    float _remainder;///<Particles that are left to be emitted from the previous time.
```
Comment (super626) [inline extensions/Particle3D/PU/CCPUEmitter.h]: As you know that the emitted particles must be int. This is the time left by last emit circle which is not enough to emit one particle.

File: extensions/Particle3D/PU/CCPUParticleSystem3D.h [hunk 9]
```diff
@@ -179,22 +196,17 @@ struct CC_DLL PUParticle3D : public Particle3D
...
+    ///< better to calculate it once at the Particle level.
     float timeFraction;
 
-        /*  Mass of a particle.
-    @remarks
-        In case of simulations where mass of a particle is needed (i.e. exploding particles of different
-        mass) this attribute can be used.
-    */
-    float mass;
+    float mass;///< mass of a particle
```
Comment (super626) [inline extensions/Particle3D/PU/CCPUParticleSystem3D.h]: The remarks can help us understand the property well. We would better try to keep them. Is there @remarks in the doxygen? You can treat it as comment of variables if not.

File: extensions/Particle3D/PU/CCPUParticleSystem3D.h [hunk 10]
```diff
@@ -203,154 +215,497 @@ struct CC_DLL PUParticle3D : public Particle3D
...
     //float widthInWorld;
     //float heightInWorld;
     //float depthInWorld;
     
 };
 
+/** @class PUParticleSystem3D
+*@brief @~english PUParticleSystem3D, the class of pu particle system, inherit from ParticleSystem3D.
+ The ParticleSystem3D,the parent of PUParticleSystem3D, is derived from Node, so one particle system can contain multiple particle systems,
+ By addChild way to mount a PUParticleSystem3D under another PUParticleSystem3D.
```
Comment (super626) [inline extensions/Particle3D/PU/CCPUParticleSystem3D.h]: You can mount a PUParticleSystem3D under another by addChild

File: extensions/Particle3D/PU/CCPUParticleSystem3D.h [hunk 11]
```diff
@@ -203,154 +215,497 @@ struct CC_DLL PUParticle3D : public Particle3D
...
+ Each particle system can be defined emitter,affector,renderer,etc. 
+ PUParticleSystem3D provides three ways to be created:
```
Comment (super626) [inline extensions/Particle3D/PU/CCPUParticleSystem3D.h]: There are three ways to create a PUParticleSystem3D

File: extensions/Particle3D/PU/CCPUParticleSystem3D.h [hunk 12]
```diff
@@ -203,154 +215,497 @@ struct CC_DLL PUParticle3D : public Particle3D
...
+    /**
+    * @~english Get the affector by name.
+    * @~chinese 根据名词获取影响器。
+    * @return @~english The name of affector. @~chinese 影响器名词。
+    */
     PUAffector* getAffector(const std::string &name);
+
+    /**
+    * @~english Get the emitter by name.
+    * @~chinese 根据名词获取发射器。
```
Comment (super626) [inline extensions/Particle3D/PU/CCPUParticleSystem3D.h]: 名字

File: extensions/Particle3D/PU/CCPUParticleSystem3D.h [hunk 13]
```diff
@@ -203,154 +215,497 @@ struct CC_DLL PUParticle3D : public Particle3D
...
 
+    /**
+    * @~english Remove all emitters.
+    * @~chinese 移除所有的发射器。
+    */
+    void removeAllEmitter();
 
+    /**
+    * @~english Add a Listener(internal method).
+    * Listener is mainly used to monitor particle emissions and disappearance.
```
Comment (super626) [inline extensions/Particle3D/PU/CCPUParticleSystem3D.h]: emission and death

**Comment by songchengjiang** [comment]
all fixed

**Pull request merged**